charts/logan/templates/ekscp-logs-configmap.yaml

```diff
@@ -0,0 +1,359 @@
+       {{- else }}
+       oci_la_entity_id "{{ $.Values.fluentd.eksControlPlane.ociLAEntityID | default $.Values.ociLAEntityID }}"
+       {{- end }}
+       {{- if $logDefinition.ociLALogSet }}
+       oci_la_log_set "{{ $logDefinition.ociLALogSet }}"
+       {{- else }}
+       oci_la_log_set "{{ $.Values.fluentd.eksControlPlane.ociLALogSet | default $.Values.ociLALogSet }}"
+       {{- end }}
+       message "${record['message']}"
+       tag ${tag}
+       </record>
+    </filter>
+    <filter eks{{- ternary (print "." $currWorker) "" $multiWorkersEnabled }}.cp.{{ $name }}.**>
+      @type record_transformer
+      enable_ruby true
+      <record>
+        oci_la_metadata ${record["oci_la_metadata"].merge({"cloudwatchloggroupname" => record["s3_key"].split("/")[0].gsub("_", "/"), "cloudwatchlogstreamname" => record["s3_key"].split("/")[2].gsub("_", "/")})}
+      </record>
+      remove_keys $.metadata
+    </filter> 
+    {{- end }}
+
+    {{- end }}
+    {{- end }}
+    {{- end }}
+
+    # Match block to ensure all the logs including concat plugin timeout logs will have same label
+    <match eks{{- ternary (print "." $currWorker) "" $multiWorkersEnabled }}.cp.**>
+       @type relabel
+       @label @NORMAL{{- ternary (print "." $currWorker) "" $multiWorkersEnabled }}
+    </match>
+
+    # Match block to set info required for oci-logging-analytics fluentd output plugin
+    <label @NORMAL{{- ternary (print "." $currWorker) "" $multiWorkersEnabled }}>
+       <match eks{{- ternary (print "." $currWorker) "" $multiWorkersEnabled }}.cp.**>
+          @type oci-logging-analytics
+          enable_ruby               true
+          namespace                 "{{ $.Values.ociLANamespace }}"
+          endpoint                  "{{ $.Values.fluentd.ociLoggingAnalyticsOutputPlugin.endpoint }}"
+          {{- if eq $authtype "config" }}
+          config_file_location      {{ $.Values.oci.path }}/{{ $.Values.oci.file }}
+          profile_name              "{{ $.Values.fluentd.ociLoggingAnalyticsOutputPlugin.profile_name }}"
+          {{- end }}
+          plugin_log_location       "{{ $.Values.fluentd.baseDir }}"
+          plugin_log_level          "{{ $.Values.fluentd.ociLoggingAnalyticsOutputPlugin.plugin_log_level }}"
+          plugin_log_file_size      "{{ $.Values.fluentd.ociLoggingAnalyticsOutputPlugin.plugin_log_file_size }}"
+          plugin_log_file_count     "{{ $.Values.fluentd.ociLoggingAnalyticsOutputPlugin.plugin_log_file_count }}"
+          # TODO
+          kubernetes_metadata_keys_mapping     {"container_name":"Container","namespace_name":"Namespace","pod_name":"Pod","container_image":"Container Image Name","host":"Node"}
+          <buffer>
+                @type                           file
+                path                            {{ $.Values.fluentd.baseDir }}/oci_la_fluentd_outplugin/logs/buffer{{- ternary (print "/worker-" $currWorker) "" $multiWorkersEnabled }}/
```

Comment:
This may collide with Fluentd running as daemonset (for all logs collection) on every node. Should we use ekscplogs instead of logs ? in the dir naming structure.

Reply:
Updated